content/docs/rfcs/36/README.md

```diff
@@ -0,0 +1,669 @@
+
+For example:
+
+```json
+{
+  "result": 0
+}
+```
+
+### `extern char* waku_peers(int nodeId)`
+
+Retrieve the list of peers known by the Waku node.
+
+**Parameters**
+
+1. `int nodeId`: The node identifier obtained from a successful execution of [`waku_new`](#extern-char-waku_newchar-configjson).
+
+**Returns**
+
+A [`JsonResponse`](#jsonresponse) containing a list of peers.
+The list of peers has this format:
+
+```json
+{
+  "result": [
+    {
+      "peerID": "16Uiu2HAmJb2e28qLXxT5kZxVUUoJt72EMzNGXB47RedcBafeDCBA",
+      "protocols": [
+        "/ipfs/id/1.0.0",
+        "/vac/waku/relay/2.0.0",
+        "/ipfs/ping/1.0.0"
+      ],
+      "addrs": [
+        "/ip4/1.2.3.4/tcp/30303"
+      ],
+      "connected": true
+    }
+  ]
+}
+```
+
+## Waku Relay
+
+### `extern char* waku_content_topic(char* applicationName, unsigned int applicationVersion, char* contentTopicName, char* encoding)`
+
+Create a content topic string according to [RFC 23](https://rfc.vac.dev/spec/23/).
+
+**Parameters**
+
+1. `char* applicationName`
+2. `unsigned int applicationVersion`
+3. `char* contentTopicName`
+4. `char* encoding`: depending on the payload, use `proto`, `rlp` or `rfc26`
+
+**Returns**
+
+`char *` containing a content topic formatted according to [RFC 23](https://rfc.vac.dev/spec/23/).
+
+```
+/{application-name}/{version-of-the-application}/{content-topic-name}/{encoding}
+```
+
+### `extern char* waku_pubsub_topic(char* name, char* encoding)`
+
+Create a pubsub topic string according to [RFC 23](https://rfc.vac.dev/spec/23/).
+
+**Parameters**
+
+1. `char* name`
+2. `char* encoding`: depending on the payload, use `proto`, `rlp` or `rfc26`
+
+**Returns**
+
+`char *` containing a content topic formatted according to [RFC 23](https://rfc.vac.dev/spec/23/).
+
+```
+/waku/2/{topic-name}/{encoding}
+```
+
+### `extern char* waku_default_pubsub_topic()`
+
+Returns the default pubsub topic used for exchanging waku messages defined in [RFC 10](https://rfc.vac.dev/spec/10/).
+
+**Returns**
+
+`char *` containing the default pubsub topic:
+
+```
+/waku/2/default-waku/proto
+```
+
+### `extern char* waku_relay_publish(int nodeId, char* messageJson, char* topic, int timeoutMs)`
+
+Publish a message using waku relay.
+
+**Parameters**
+
+1. `int nodeId`: The node identifier obtained from a successful execution of [`waku_new`](#extern-char-waku_newchar-configjson).
+2. `char* messageJson`: JSON string containing the [Waku Message](https://rfc.vac.dev/spec/14/) as [`JsonMessage`](#JsonMessage).
+3. `char* topic`: pubsub topic on which to publish the message.
+   If `NULL`, it uses the default pubsub topic.
+4. `int timeoutMs`: Timeout value in milliseconds to execute the call.
+   If the function execution takes longer than this value,
+   the execution will be canceled and an error returned.
+   Use `0` for no timeout.
+
+**Returns**
+
+A [`JsonResponse`](#jsonresponse).
+If the execution is successful, the `result` field contains the message ID.
```

Comment:
What is a message id?

Reply:
The messageId is the sha2-256 hash of the data sent via gossipsub (the wakumessage) that is generated in the messageIdProvider. In go-waku when I broadcast a message I return this hash. 

(I found it useful for status-go in particular since we store messages locally in the DB and they're identified with this hash)

Reply:
Do we want to standardize the message id concept? ie, specify its generation?

Cc @jm-clius

Reply:
Not sure we want to/need to be too strict here in terms of how the message ID gets generated. IMO the only contract w the API should be that the message ID is returned in the response, whichever way that node generates the ID.